Kconfig.zephyr

```diff
@@ -738,6 +738,18 @@ endmenu
 
 menu "Boot Options"
 
+config APPLICATION_VERSION
```

Comment:
`APPLICATION_VERSION` is to generic for something that is related to mcuboot, it should be something like `MCUBOOT_APPLICATION_VERSION`

Reply:
It can be used by anything and is not mcuboot-centric, e.g. the binary descriptor PR that has been brought up in the architecture working group might use it.

Reply:
And don't apply any formatting requirement on this.